src/session/parse_enum.rs

```diff
@@ -0,0 +1,61 @@
+use crate::session::UnknownEnumValueError;
+
+pub trait ProtoEnum<Proto> {
```

Comment:
I am not sold on this trait. What is the problem is it trying to solve?

Reply:
You can see more context in the PR: https://github.com/google/rrg/pull/42. For multiple protobuf enums we need to:
- know the default value (which is not a property of enum itself, but of the structure which uses the enum),
- be able to parse it easily,
- be able to detect that the received value of the enum is not recognised and return an approrpiate error.

With example implementation:
```
impl ProtoEnum<ActionType> for ActionType {
    fn default() -> Self {
        FileFinderAction::default().action_type()
    }
    fn from_i32(val: i32) -> Option<Self> {
        ActionType::from_i32(val)
    }
}
```

You can parse the enum from proto just by:

```
parse_enum::<ActionType>(proto.action_type)?;
```

Reply:
Small thing to add: in most of cases you parse by just `parse_enum(proto.action_type)` - no type annotations are needed.